probe_scraper/scrapers/git_scraper.py

```diff
@@ -41,15 +41,32 @@
     "engine-gecko-beta": [
         "9bd9d7fa6c679f35d8cbeb157ff839c63b21a2e6"  # Missing schema update from v1 to v2
     ],
+    "firefox-desktop": [
+        "43d8cf138695faae2fca0adf44c94f47fdadfca8"  # Missing gfx/metrics.yaml
+    ]
 }
 
 
+def _file_in_repo_head(repo, filename):
+    # adapted from https://stackoverflow.com/a/25961128
+    subtree = repo.head.commit.tree
+    for path_element in os.path.dirname(filename).split(os.path.sep):
+        try:
+            subtree = subtree[path_element]
+        except KeyError:
+            return False  # subdirectory not in tree
+    return filename in tree
+
+
 def get_commits(repo, filename):
     sep = ":"
     log_format = '--format="%H{}%ct"'.format(sep)
-    change_commits = enumerate(repo.git.log(log_format, filename).split("\n"))
-    most_recent_commit = enumerate(repo.git.log("-n", "1", log_format).split("\n"))
-    commits = set(change_commits) | set(most_recent_commit)
+    # include "--" to prevent error for filename not in current tree
+    change_commits = repo.git.log(log_format, "--", filename).split("\n")
+    commits = set(enumerate(change_commits))
+    if _file_in_repo_head(repo, filename):
+        # include HEAD when it contains filename
+        commits |= set(enumerate(repo.git.log("-n", "1", log_format).split("\n")))
```

Comment:
This is needed because `SKIP_COMMITS` doesn't allow `HEAD`.

Also, this can't be used to also filter `change_commits` because for reasons I don't understand `_file_in_repo_head` doesn't work with `subtree = repo.tree(commit)` to check arbitrary commits.